### PR TITLE
L1T: Updates for histogrammed PF jets & firmware emulator comparisons

### PR DESCRIPTION
**Pull request by bundocka**
- Update to histogrammed PF jet algo config
- Add pattern generator and comparison script for fw vs emu comparisons

### CODE REVIEW

File: L1Trigger/L1CaloTrigger/plugins/Phase1L1TJetProducer.cc [hunk 1]
```diff
@@ -159,6 +159,18 @@ void Phase1L1TJetProducer::produce(edm::Event& iEvent, const edm::EventSetup& iS
     // dumping the data
     this -> _caloGrid -> Reset();
     this -> _fillCaloGrid<>(*(this -> _caloGrid), *inputCollectionHandle);
+
```
Comment (simonecid) [inline L1Trigger/L1CaloTrigger/plugins/Phase1L1TJetProducer.cc:162]: Line 162 to 173 do not actually do anything, I would comment the entire block, not only the content of the double for-loop

File: L1Trigger/L1CaloTrigger/plugins/Phase1L1TJetProducer.cc [hunk 2]
```diff
@@ -351,9 +368,11 @@ template <class Container>
 void Phase1L1TJetProducer::_fillCaloGrid(TH2F & caloGrid, const Container & triggerPrimitives)
 {
   //Filling the calo grid with the primitives
-  for (auto primitiveIterator = triggerPrimitives.begin(); primitiveIterator != triggerPrimitives.end(); primitiveIterator++) 
-  {
-    caloGrid.Fill((float) primitiveIterator -> eta(), (float) primitiveIterator -> phi(), (float) primitiveIterator -> pt());
+  for (auto primitiveIterator = triggerPrimitives.begin(); primitiveIterator != triggerPrimitives.end(); primitiveIterator++){
```
Comment (simonecid) [inline L1Trigger/L1CaloTrigger/plugins/Phase1L1TJetProducer.cc]: There is a double curly bracket layer here. To be consistent with the coding style used in other parts of the code, I would remove the curly bracket at the end of line 371 and at line 375 and then reduce by a tabulation level.

**Review by simonecid — CHANGES_REQUESTED** [review]
Looks OK to me. Two changes are required for cleanness: commenting a code block that does not do anything, turning a double curly bracket ({{ .... }}) in a single one ({ .... }).

Simone

File: L1Trigger/L1CaloTrigger/plugins/Phase1L1TJetProducer.cc (hunk 2 above)
Comment (bundocka) [reply L1Trigger/L1CaloTrigger/plugins/Phase1L1TJetProducer.cc]: https://github.com/cms-l1t-offline/cmssw/pull/803/commits/a9d12f59c95bf20212263d403bbf6b794f54be08

File: L1Trigger/L1CaloTrigger/plugins/Phase1L1TJetProducer.cc (hunk 1 above)
Comment (bundocka) [reply L1Trigger/L1CaloTrigger/plugins/Phase1L1TJetProducer.cc:162]: https://github.com/cms-l1t-offline/cmssw/pull/803/commits/a9d12f59c95bf20212263d403bbf6b794f54be08

**Review by simonecid — APPROVED** [review]

**Pull request merged**